src/runaqa.ts

```diff
@@ -1,324 +1,388 @@
-      )
-    }
-
-    if (fs.existsSync('/usr/bin/apt-get')) {
-      //disable apport
-      await exec.exec('sudo service apport stop')
-    }
-  }
-}
-
-function setSpec(): void {
-  if (IS_WINDOWS) {
-    process.env['SPEC'] = 'win_x86-64_cmprssptrs'
-  } else if (process.platform === 'darwin') {
-    process.env['SPEC'] = 'osx_x86-64_cmprssptrs'
-  } else {
-    process.env['SPEC'] = 'linux_x86-64_cmprssptrs'
-  }
-}
-
-async function getAqaTestsRepo(aqatestsRepo: string): Promise<void> {
-  let repoBranch = ['adoptium/aqa-tests', 'master']
-  if (aqatestsRepo.length !== 0) {
-    repoBranch = parseRepoBranch(aqatestsRepo)
-  }
-  await exec.exec(
-    `git clone --depth 1 -b ${repoBranch[1]} https://github.com/${repoBranch[0]}.git`
-  )
-  process.chdir('aqa-tests')
-}
-
-function getAqaSystemTestsRepo(aqasystemtestsRepo: string): void {
-  const repoBranch = parseRepoBranch(aqasystemtestsRepo)
-  process.env.ADOPTOPENJDK_SYSTEMTEST_REPO = repoBranch[0]
-  process.env.ADOPTOPENJDK_SYSTEMTEST_BRANCH = repoBranch[1]
-}
-
-async function runGetSh(
-  tkgRepo: string,
-  openj9Repo: string,
-  vendorTestParams: string,
-  jdksource: string,
-  customizedSdkUrl: string,
-  sdkdir: string
-): Promise<void> {
-  let parameters = ''
-  if (tkgRepo.length !== 0) {
-    const repoBranch = parseRepoBranch(tkgRepo)
-    parameters += `--tkg_branch ${repoBranch[1]} --tkg_repo https://github.com/${repoBranch[0]}.git`
-  }
-  if (openj9Repo.length !== 0) {
-    const repoBranch = parseRepoBranch(openj9Repo)
-    parameters += ` --openj9_branch ${repoBranch[1]} --openj9_repo https://github.com/${repoBranch[0]}.git`
-  }
-  if (jdksource.length !== 0) {
-    parameters += ` --sdk_resource ${jdksource}`
-  }
-  if (customizedSdkUrl.length !== 0) {
-    parameters += ` --customizedURL ${customizedSdkUrl}`
-  }
-  if (sdkdir.length !== 0) {
-    parameters += ` --sdkdir ${sdkdir}`
-  }
-  if (IS_WINDOWS) {
-    await exec.exec(`bash ./get.sh ${parameters} ${vendorTestParams}`)
-  } else {
-    await exec.exec(`./get.sh ${parameters} ${vendorTestParams}`)
-  }
-}
-
-function parseRepoBranch(repoBranch: string): string[] {
-  const tempRepo = repoBranch.replace(/\s/g, '')
-  var slashIndexCheck = tempRepo.indexOf( "/" )
-  var colonIndexCheck = tempRepo.indexOf( ":" )
-  if(slashIndexCheck>0 && colonIndexCheck>0 && slashIndexCheck<colonIndexCheck) {
-    return tempRepo.split(':')
-  } else  {
-    return "Error in string parameter format. Required form: 'octocat/projectnames:branch' "
-  }
-}
+import * as exec from '@actions/exec'
+import * as core from '@actions/core'
+import * as io from '@actions/io'
+import * as tc from '@actions/tool-cache'
+import * as path from 'path'
+import * as fs from 'fs'
+import {ExecOptions} from '@actions/exec/lib/interfaces'
+
+let tempDirectory = process.env['RUNNER_TEMP'] || ''
+const IS_WINDOWS = process.platform === 'win32'

```

Comment:
Is it always win32?  Can it be win64?

Reply:
This is always win32. Its a Node.js functionality. 

From the docs:

```
Currently possible values are:

'aix'
'darwin'
'freebsd'
'linux'
'openbsd'
'sunos'
'win32'


https://nodejs.org/api/process.html#processplatform